src/index.js

```diff
@@ -1,10 +1,292 @@
-// This is the JavaScript entry file - your code begins here
-// Do not delete or rename this file ********
 
-// An example of how you tell webpack to use a CSS (SCSS) file
 import './css/index.scss';
+import User from './User';
+import Guest from './Guest';
+import Hotel from './Hotel';
+import { apiRequest } from './fetchAPIData';
 
-// An example of how you tell webpack to use an image (also need to link to it in the index.html)
-import './images/turing-logo.png'
+let customerData;
+let bookingData;
+let roomData;
+let currentGuest;
+let hotel;
 
-console.log('This is the JavaScript entry file - your code begins here.');
+
+const guestSearchBar = document.getElementById('roomForm');
+const filterOption = document.getElementById('roomTypes');
+const searchByDateSection = document.getElementById('searchSection');
+const avaiableRoomCard = document.getElementById('roomCards');
+
+window.addEventListener('load', getAllAPIData);
+guestSearchBar.addEventListener('click', handleGuestSearchClick);
+searchByDateSection.addEventListener('click', handleSearchByDate);
+avaiableRoomCard.addEventListener('click', handleBookRoom)
+
+
+function getAllAPIData() {
+  const customers = apiRequest.getCustomerData();
+  const bookings = apiRequest.getBookingData();
+  const rooms = apiRequest.getRoomData();
+  Promise.all([customers, bookings, rooms]).then(data => {
+    assignAPIData(data[0], data[1], data[2]);
+  })
+}
+
+function assignAPIData(customers, bookings, rooms) {
+  customerData = customers;
+  bookingData = bookings;
+  roomData = rooms;
+  hotel = new Hotel(roomData, bookingData, '2021/03/07');
+  createGuest();
+}
+
+function createGuest() {
+  currentGuest = new Guest('customer1', customerData);
+  activateGuestMethods();
+  displayGuestDashboard();
+}
+
+function activateGuestMethods() {
```

Comment:
maybe a rename on this to be more readable..like createGuestData() or updateGuestRecords()